From dca6005cf69d8d11e04401c187a322e10b74b8fb Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Sun, 5 May 2019 17:36:57 +0300
Subject: [PATCH 0258/1239] arm64: mvebu: dts: fix efuse entries and OTP base
 addresses

Change the CP LD efuse names to avoid naming collisions when
more than one CP is connected to AP.
Replace OTP address with offset - the base OTP address should
be calculated and added by the efuse driver
Update A38x DTSI for reflecting OTP address changes

Change-Id: Iad16301dba90e858b2e0c2613cd22275342c16f3
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
---
 arch/arm/dts/armada-38x.dtsi   | 2 +-
 arch/arm/dts/armada-ap80x.dtsi | 6 +++---
 arch/arm/dts/armada-cp110.dtsi | 8 ++++----
 3 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/arch/arm/dts/armada-38x.dtsi b/arch/arm/dts/armada-38x.dtsi
index 3193070da7..28151f61ac 100644
--- a/arch/arm/dts/armada-38x.dtsi
+++ b/arch/arm/dts/armada-38x.dtsi
@@ -594,7 +594,7 @@
 			efuse: efuse@f9000 {
 				compatible = "marvell,mvebu-fuse-hd";
 				reg = <0xe4008 0x4>;
-				otp-mem = <0xf60f9000>;
+				otp-mem = <0xf9000>;
 				rows-count = <64>;
 				status = "disabled";
 			};
diff --git a/arch/arm/dts/armada-ap80x.dtsi b/arch/arm/dts/armada-ap80x.dtsi
index 43def8e55c..51bb6082be 100644
--- a/arch/arm/dts/armada-ap80x.dtsi
+++ b/arch/arm/dts/armada-ap80x.dtsi
@@ -121,21 +121,21 @@
 			ap_ld_efuse0: efuse-0@6F8F00 {
 				compatible = "marvell,mvebu-fuse-ld-user";
 				reg = <0x6F8008 0x4>;
-				otp-mem = <0xF06F8F00>;
+				otp-mem = <0x6F8F00>;
 				status = "disabled";
 			};
 
 			ap_ld_efuse1: efuse-1@6F8F00 {
 				compatible = "marvell,mvebu-fuse-ld-prop";
 				reg = <0x6F8008 0x4>;
-				otp-mem = <0xF06F8F00>;
+				otp-mem = <0x6F8F00>;
 				status = "disabled";
 			};
 
 			ap_hd_efuse0: efuse@6F9000  {
 				compatible = "marvell,mvebu-fuse-hd";
 				reg = <0x6F8008 0x4>;
-				otp-mem = <0xF06F9000>;
+				otp-mem = <0x6F9000>;
 				rows-count = <64>;
 				status = "disabled";
 			};
diff --git a/arch/arm/dts/armada-cp110.dtsi b/arch/arm/dts/armada-cp110.dtsi
index 7f94538843..f3488b404e 100644
--- a/arch/arm/dts/armada-cp110.dtsi
+++ b/arch/arm/dts/armada-cp110.dtsi
@@ -55,17 +55,17 @@
 				status = "okay";
 			};
 
-			CP110_LABEL(ld_efuse0): efuse0@400F00 {
+			CP110_LABEL(ld_efuse0): CP110_LABEL(efuse0)@400F00 {
 				compatible = "marvell,mvebu-fuse-ld-user";
 				reg = <0x400008 0x4>;
-				otp-mem = <0xF2400F00>;
+				otp-mem = <0x400F00>;
 				status = "disabled";
 			};
 
-			CP110_LABEL(ld_efuse1): efuse1@400F00 {
+			CP110_LABEL(ld_efuse1): CP110_LABEL(efuse1)@400F00 {
 				compatible = "marvell,mvebu-fuse-ld-prop";
 				reg = <0x400008 0x4>;
-				otp-mem = <0xF2400F00>;
+				otp-mem = <0x400F00>;
 				status = "disabled";
 			};
 
-- 
2.29.0

